apps/wallet/src/ui/app/staking/stake/StakingCard.tsx

```diff
@@ -136,17 +137,17 @@ function StakingCard() {
     const [coinDecimals] = useCoinDecimals(coinType);
     const [gasDecimals] = useCoinDecimals(GAS_TYPE_ARG);
     const maxSuiSingleCoinBalance = useIndividualCoinMaxBalance(SUI_TYPE_ARG);
-
+    const gasBudget = useGasBudgetInMist(DEFAULT_GAS_BUDGET_FOR_STAKE);
     //TODO: this is a hack to get the total amount of gas coins
     // for unstake add gas
+    //NOTE: I think we should remove this. Validation expects just the total amount of SUI
     const gasAggregateBalance = useMemo(
         () =>
             unstake
-                ? coinBalance + BigInt(DEFAULT_GAS_BUDGET_FOR_STAKE)
+                ? coinBalance + BigInt(gasBudget.gasBudget || 0)
```

Comment:
do we need this `|| 0` check every time we use `gasBudget` ?

Reply:
can we just `let gasBudget = 0` in the hook instead of `null` ?

Reply:
Good point, I think I use it to check if query is loading, so if null means loading but maybe I should follow the convention from useQuery to be undefined. I will change it to undefined

Reply:
Change it